src/ifd.rs

```diff
@@ -504,12 +533,24 @@ impl ImageFileDirectory {
         }
     }
 
-    pub async fn get_tile(&self, x: usize, y: usize) {
+    pub async fn get_tile(
```

Comment:
Small NIT but I'd recommend breaking this up into two separate methods, something like `get_tile` and `get_and_decode_tile`.  The first would parse the header and send a range request for the desired offset, the second is a wrapper around the first and `decode_tile`.

This does a better job decoupling IO/CPU bound code by allowing the caller to easily perform the range request without also decompressing the data.

Reply:
Yes, that makes a lot of sense. In general, optimizing IO/CPU handling is something I want to leave for future PRs after we have the general prototype working and some tests passing.